zerver/migrations/0023_subscription_flags.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import models, migrations
+import bitfield.models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('zerver', '0022_subscription_pin_to_top'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='subscription',
+            name='flags',
+            field=bitfield.models.BitField([b'stream_admin', b'moderator', b'regular', b'read_only'], default=0),
```

Comment:
We should call this something more descriptive than `flags`.

Also, we may want to model this as sets of permissions instead of the user's role. So, for instance, instead of a `read_only` user role, we would have permissions like `can_write`, `can_read`, `can_moderate`, `can_add_remove_users`, etc... @timabbott  thoughts?